tests/phpunit/tests/admin/wpListTable.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/**
+ * @group admin
+ *
+ * @covers WP_List_Table
+ */
+class Tests_Admin_WpListTable extends WP_UnitTestCase {
+
+	/**
+	 * Tests that `WP_List_Table::get_column_info()` only adds the primary
+	 * column header when necessary.
+	 *
+	 * @ticket 34564
+	 *
+	 * @dataProvider data_should_only_add_primary_column_when_needed
+	 *
+	 * @covers WP_List_Table::get_column_info
+	 *
+	 * @param string $list_class          The name of the WP_List_Table child class.
+	 * @param array  $headers             A list of column headers.
+	 * @param array  $expected            The expected column headers.
+	 * @param int    $expected_hook_count The expected number of times the hook is called.
+	 */
+	public function test_should_only_add_primary_column_when_needed( $list_class, $headers, $expected, $expected_hook_count ) {
+		$hook = new MockAction();
+		add_filter( 'list_table_primary_column', array( $hook, 'filter' ) );
+
+		// This is reset between tests.
+		$GLOBALS['hook_suffix'] = 'my-hook';
+
+		$list_table = _get_list_table( $list_class );
```

Comment:
```suggestion
		$list_table = _get_list_table( $list_class, array( 'screen' => 'test-screen' ) );
```
Passing a `screen` value in the args removes the need to set the `hook_suffix` global.